> 4. I suspect this do not works 1 to 1 with regarding to OVERRIDE_GIT_DESCRIBE, I think adding back logic to handle shorter cases might be needed. Test cases are basically running this with `export OVERRIDE_GIT_DESCRIBE=v2.3.4` or `export OVERRIDE_GIT_DESCRIBE=v2.3.4-0-g0123456789` or with something like `export OVERRIDE_GIT_DESCRIBE=v2.3.4-100` or `export OVERRIDE_GIT_DESCRIBE=v2.3.4-100-gabcdabcdab`. First two cases should generate the same wheels, second pair of cases should also generate the same wheels.

- `export OVERRIDE_GIT_DESCRIBE=v2.3.4`
```
duckdb-2.3.4.tar.gz
```
```
Processing ./duckdb-2.3.4.tar.gz
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: duckdb
  Building wheel for duckdb (pyproject.toml) ... done
  Created wheel for duckdb: filename=duckdb-2.3.4-cp311-cp311-macosx_14_0_arm64.whl size=14833119 sha256=66809d4a3cd602bb897a73377b1764ff8a7fe651ffdae50ee06280cac06369cd
  Stored in directory: /Users/thijs/Library/Caches/pip/wheels/e6/c1/08/74928ea55dc362739b095350908d70d8de8dc543f4fe89481d
Successfully built duckdb
Installing collected packages: duckdb
Successfully installed duckdb-2.3.4
```
```
>>> duckdb.__version__
'2.3.4'
```
- `export OVERRIDE_GIT_DESCRIBE=v2.3.4-0-g0123456789`
```
duckdb-2.3.4.tar.gz
```
```
Processing ./duckdb-2.3.4.tar.gz
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: duckdb
  Building wheel for duckdb (pyproject.toml) ... done
  Created wheel for duckdb: filename=duckdb-2.3.4-cp311-cp311-macosx_14_0_arm64.whl size=14833137 sha256=095661297f0457e45679dfb11e84bcda9859103b150a0d5470507c926735e934
  Stored in directory: /Users/thijs/Library/Caches/pip/wheels/e6/c1/08/74928ea55dc362739b095350908d70d8de8dc543f4fe89481d
Successfully built duckdb
Installing collected packages: duckdb
Successfully installed duckdb-2.3.4
```
```
>>> duckdb.__version__
'2.3.4'
```
- `export OVERRIDE_GIT_DESCRIBE=v2.3.4-100`
(leaving out the install, because it takes looong, instead I'll post the defines in `pragma_version.cpp` in the produced tar)
```
duckdb-2.4.0.dev100.tar.gz
```
```
#ifndef DUCKDB_PATCH_VERSION
#define DUCKDB_PATCH_VERSION "4-dev100"
#endif
#ifndef DUCKDB_MINOR_VERSION
#define DUCKDB_MINOR_VERSION 4
#endif
#ifndef DUCKDB_MAJOR_VERSION
#define DUCKDB_MAJOR_VERSION 2
#endif
#ifndef DUCKDB_VERSION
#define DUCKDB_VERSION "v2.4.4-dev100"
#endif
#ifndef DUCKDB_SOURCE_ID
#define DUCKDB_SOURCE_ID "d2bc4b0ac4"
#endif
```
- `export OVERRIDE_GIT_DESCRIBE=v2.3.4-100-gabcdabcdab`
```
duckdb-2.4.0.dev100.tar.gz
```
```
#ifndef DUCKDB_PATCH_VERSION
#define DUCKDB_PATCH_VERSION "4-dev100"
#endif
#ifndef DUCKDB_MINOR_VERSION
#define DUCKDB_MINOR_VERSION 4
#endif
#ifndef DUCKDB_MAJOR_VERSION
#define DUCKDB_MAJOR_VERSION 2
#endif
#ifndef DUCKDB_VERSION
#define DUCKDB_VERSION "v2.4.4-dev100"
#endif
#ifndef DUCKDB_SOURCE_ID
#define DUCKDB_SOURCE_ID "abcdabcdab"
#endif
```

Some extra while I'm at it:
```
➜  pythonpkg git:(python_use_version_file) ✗ export OVERRIDE_GIT_DESCRIBE=0                                                                 
➜  pythonpkg git:(python_use_version_file) ✗ python3 setup.py sdist                                                                         
Traceback (most recent call last):
  File "/Users/thijs/DuckDBLabs/duckdb/tools/pythonpkg/setup.py", line 253, in <module>
    (source_list, include_list, original_sources) = package_build.build_package(
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/thijs/DuckDBLabs/duckdb/tools/pythonpkg/../../scripts/package_build.py", line 330, in build_package
    '#define DUCKDB_MINOR_VERSION {}'.format(int(dev_v_parts[1])),
                                                 ~~~~~~~~~~~^^^
IndexError: list index out of range
```

```
➜  pythonpkg git:(python_use_version_file) ✗ export OVERRIDE_GIT_DESCRIBE=v0
➜  pythonpkg git:(python_use_version_file) ✗ python3 setup.py sdist         
Traceback (most recent call last):
  File "/Users/thijs/DuckDBLabs/duckdb/tools/pythonpkg/setup.py", line 253, in <module>
    (source_list, include_list, original_sources) = package_build.build_package(
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/thijs/DuckDBLabs/duckdb/tools/pythonpkg/../../scripts/package_build.py", line 330, in build_package
    '#define DUCKDB_MINOR_VERSION {}'.format(int(dev_v_parts[1])),
                                                 ~~~~~~~~~~~^^^
IndexError: list index out of range
```

```
➜  pythonpkg git:(python_use_version_file) ✗ export OVERRIDE_GIT_DESCRIBE=v0.0
➜  pythonpkg git:(python_use_version_file) ✗ python3 setup.py sdist           
Traceback (most recent call last):
  File "/Users/thijs/DuckDBLabs/duckdb/tools/pythonpkg/setup.py", line 253, in <module>
    (source_list, include_list, original_sources) = package_build.build_package(
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/thijs/DuckDBLabs/duckdb/tools/pythonpkg/../../scripts/package_build.py", line 336, in build_package
    '#define DUCKDB_PATCH_VERSION "{}"'.format(dev_v_parts[2]),
                                               ~~~~~~~~~~~^^^
IndexError: list index out of range
```

`export OVERRIDE_GIT_DESCRIBE=v0.0.1`
```
duckdb-0.0.1.tar.gz
```
```
#ifndef DUCKDB_PATCH_VERSION
#define DUCKDB_PATCH_VERSION "1"
#endif
#ifndef DUCKDB_MINOR_VERSION
#define DUCKDB_MINOR_VERSION 0
#endif
#ifndef DUCKDB_MAJOR_VERSION
#define DUCKDB_MAJOR_VERSION 0
#endif
#ifndef DUCKDB_VERSION
#define DUCKDB_VERSION "v0.0.1"
#endif
#ifndef DUCKDB_SOURCE_ID
#define DUCKDB_SOURCE_ID "d2bc4b0ac4"
#endif
```

--- a/scripts/amalgamation.py
+++ b/scripts/amalgamation.py
@@ -277,7 +277,7 @@ def git_commit_hash():
 # - scripts/package_build.py
 # - tools/pythonpkg/setup.py
 ######
-main_branch_versioning = os.getenv('MAIN_BRANCH_VERSIONING') or 1
+main_branch_versioning = False if os.getenv('MAIN_BRANCH_VERSIONING') == "0" else True
 
 
 def git_dev_version():
@@ -291,7 +291,7 @@ def git_dev_version():
         else:
             # not on a tag: increment the version by one and add a -devX suffix
             # this needs to keep in sync with changes to CMakeLists.txt
-            if main_branch_versioning:
+            if main_branch_versioning == True:
                 # increment minor version
                 version_splits[1] = str(int(version_splits[1]) + 1)
             else:

